iroh-sync/src/ranger.rs

```diff
@@ -238,30 +237,30 @@ pub trait Store<E: RangeEntry>: Sized {
         E: 'a;
 
     /// Get a the first key (or the default if none is available).
-    fn get_first(&self) -> Result<E::Key, Self::Error>;
+    fn get_first(&mut self) -> Result<E::Key, Self::Error>;
 
     /// Get a single entry.
-    fn get(&self, key: &E::Key) -> Result<Option<E>, Self::Error>;
+    fn get(&mut self, key: &E::Key) -> Result<Option<E>, Self::Error>;
 
     /// Get the number of entries in the store.
-    fn len(&self) -> Result<usize, Self::Error>;
+    fn len(&mut self) -> Result<usize, Self::Error>;
 
     /// Returns `true` if the vector contains no elements.
-    fn is_empty(&self) -> Result<bool, Self::Error>;
+    fn is_empty(&mut self) -> Result<bool, Self::Error>;
 
     /// Calculate the fingerprint of the given range.
-    fn get_fingerprint(&self, range: &Range<E::Key>) -> Result<Fingerprint, Self::Error>;
+    fn get_fingerprint(&mut self, range: &Range<E::Key>) -> Result<Fingerprint, Self::Error>;
 
     /// Insert the given key value pair.
-    fn put(&mut self, entry: E) -> Result<(), Self::Error>;
+    fn kv_put(&mut self, entry: E) -> Result<(), Self::Error>;
```

Comment:
I'd prefer `entry_put` here instead of introducing a new `kv` term.

Reply:
the intention was to distinguish between the "willow put" and the (currently underlying) kv store style put.